oauthproxy.go

```diff
@@ -484,6 +487,13 @@ func (p *OAuthProxy) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 }
 
 func (p *OAuthProxy) serveHTTP(rw http.ResponseWriter, req *http.Request) {
+	var ctx context.Context
```

Comment:
I think this should be in our middleware package (`pkg/middleware`) and added to our request middleware here: https://github.com/oauth2-proxy/oauth2-proxy/blob/220b3708fc4e1de5e30135b48c8d8a2ef7efd518/oauthproxy.go#L232

Whether the logic of the middleware is in the tracing implementation middleware itself, or split out into your separate tracing package is up to you.

Then we have tracing as part of our `req.Context()` here if needed and we don't need to change all the signatures below to add a `context.Context`